packages/web3-eth-accounts/src/wallet.ts

```diff
@@ -157,10 +157,12 @@ export class Wallet<
 		if (typeof account === 'string') {
 			return this.add(this._accountProvider.privateKeyToAccount(account));
 		}
-
-		const index = this.length;
+		let index = this.length;
+		const existAccount = this.get(account.address);
+		if (existAccount) {
+			index = this._addressMap.get(account.address.toLowerCase()) ?? index;
+		}
 		this._addressMap.set(account.address.toLowerCase(), index);
-
 		this[index] = account;
```

Comment:
I think there is here a chance to have a doublication if the passed account is of type `string`.
And here is a suggested draft to overcome that:

```suggestion
		let accountToAdd =
			typeof account === 'string'
				? this._accountProvider.privateKeyToAccount(account)
				: account;
		let index = this.length;
		const existAccount = this.get(accountToAdd.address);
		if (existAccount) {
			index = this._addressMap.get(accountToAdd.address.toLowerCase()) ?? index;
		}
		this._addressMap.set(accountToAdd.address.toLowerCase(), index);
		this[index] = accountToAdd;
```

Reply:
This is not my code. But this code uses recursion if the type is a string. So, your suggestion also can be placed. Logic is the same.

Reply:
Yes, it is not part of the scope of this MR. However, I suggested it to be part of your suggested enhancements as you checked for duplication if the account is an object, but did not when the account is a `string`. So, my suggestion was basically to expand the new `if` that you added. :smile: 
And you are right, it is not necessarily needed to be handled in this MR. So, I created an issue for this bug so handle it later: https://github.com/ChainSafe/web3.js/issues/5388 :smile: